stock_picking_invoice_link/models/stock_move.py

```diff
@@ -33,4 +33,29 @@ def write(self, vals):
             for move in self:
                 if move.state == "done" and move.invoice_line_ids:
                     raise UserError(_("You can not modify an invoiced stock move"))
-        return super().write(vals)
+        res = super().write(vals)
+        if vals.get("state", "") == "done":
+            stock_moves = self.get_moves_delivery_link_invoice()
+            for stock_move in stock_moves.filtered(
+                lambda sm: sm.sale_line_id and sm.product_id.invoice_policy == "order"
+            ):
+                inv_type = stock_move.to_refund and "out_refund" or "out_invoice"
```

Comment:
@CarlosRoca13 Missing the field to compare to 'out_refund_invoice' in your condition

Reply:
It's an assignation @rousseldenis... `inv_type` will take `"out_refund"` value if `stock_move.to_refund` is True, if it is False `inv_type` will take "out_invoice"